docs/node-js-notes.md

```diff
@@ -0,0 +1,63 @@
+# Using Node.js with your project
+
+As you work with the sample code in this repo, you will find references to `@brightsign` JavaScript modules. These modules provide BrightSign specific functionality and are built in to the device firmware. These modules can be integrated with your code to provide rich experiences on the device.
+
+There are two main ways to run Node.js on your Brightsign device.
+
+# roHtmlWidget
+
+Create an `roHtmlWidget` object using BrightScript in your `autorun.brs` with the `nodejs_enabled` flag set to `true`. This creates a WebKit HTML renderer instance that can access the BrightSign JS modules. Set the URL of the widget to be an externally hosted page or to a file on your local filesystem. You can then set the size and position of the widget to be displayed on your device. This essentially creates a floating browser window that displays content of your choice.
+
+```
+aa=createobject("roassociativearray")
+aa.url="http://test-server/index.html"
+aa.nodejs_enabled=true
+
+r=createobject("rorectangle",0,0,1920,1080)
+h=createobject("rohtmlwidget",r,aa)
+h.show()
+```
+
+Keep in mind that any JavaScript code in your `roHtmlWidget` will only be run as long as the widget is active. You can create and display multiple instances at once.
+
+![roHtmlWidget](./images/rohtmlwidget.png)
+
+
+# roNodeJs
+Launch a Node.js process in the background in BrightScript and supply any arguments you might need. Your code will need to be on your local filesystem.
+
+```
+node = CreateObject("roNodeJs", "index.js", {message_port:my_message_port, node_arguments: ["arg"], arguments: ["arg1", "arg2"]})
+```
+
+Unlike `roHtmlWidget`, an `roNodeJs` object will run in the background uninterrupted. You can use this for long running processes like gathering metrics or running a web server. You can even pass messages between this process and code running in an `roHtmlWidget` using an `roNodeJsEvent`.
```

Comment:
This is not entirely the case - what's missing is the message would be received in brightscript's roMessagePort as an roNodeJsEvent and if it needed to be sent to roHtmlWidget, a new roMessagePort.SendJsMessage() must be executed. 

See, https://brightsign.atlassian.net/wiki/spaces/DOC/pages/404621871/roNodeJsEvent and https://brightsign.atlassian.net/wiki/spaces/DOC/pages/404619518/messageport for how that would be implemented. 

Though, I believe a direct "pass" from the roNodeJs object using the following may be possible. A quick test may be needed. Or an example could be created to explain how to pass messages between multiple chromium instances. 
```
messagePort.SendJsMessage(...{destination: "roHtmlWidget_mywidget"})
```

and roHtmlWidget has an event listener for the messageport `bsmessage` event, 
```
var MESSAGE_PORT = require("@brightsign/messageport");
var bsMessage = new MESSAGE_PORT();

bsMessage.addEventListener('bsmessage', function(msg) { 
if msg.destination === roHtmlWidget_mywidget console.log("Found it")
 }) 
```